### PR TITLE
fix test_request_snapshot_matched_change

### PR DESCRIPTION
**Pull request by Fullstop000**
Fix https://github.com/pingcap/raft-rs/issues/274

Signed-off-by: Fullstop000 <fullstop1005@gmail.com>

### CODE REVIEW

**Comment by Fullstop000** [comment]
PTAL @overvenus  @hicqu

File: harness/tests/integration_cases/test_raft.rs [hunk 1]
```diff
@@ -4495,7 +4495,16 @@ fn test_request_snapshot_matched_change() {
...
         nt.peers.get_mut(&1).unwrap().tick();
     }
-    let msg_hb = nt.peers.get_mut(&1).unwrap().msgs.pop().unwrap();
+    let msg_hb = nt
+        .peers
+        .get_mut(&1)
+        .unwrap()
+        .msgs
+        .iter()
+        .cloned()
```
Comment (Hoverbear) [inline harness/tests/integration_cases/test_raft.rs]: It seems like we're cloning twice here?

**Comment by brson** [comment]
PTAL @Hoverbear @nrc

**Comment by Hoverbear** [comment]
Gosh you are so awesome 😍

**Pull request merged**